### PR TITLE
[TASK] Remove instead of add filter twice

### PR DESCRIPTION
**Pull request by xf-**
## Description

Instead of adding filter twice - remove the filter.

## Checklist:

- [x] My code follows the code style of this project. (CI will test it anyway and also needs approval)

### CODE REVIEW

File: lib/filters/genericnode.js [hunk 1]
```diff
@@ -52,7 +52,8 @@ define(['helper'], function (helper) {
     return {
       run: run,
       setRefresh: setRefresh,
-      render: render
+      render: render,
+      unique: value + name
```
Comment (xf-) [inline lib/filters/genericnode.js]: @genofire i got no name and value to compare them. If you have a better idea, you are welcome :)

File: lib/datadistributor.js [hunk 2]
```diff
@@ -54,10 +54,21 @@ define(['filters/nodefilter'], function (NodeFilter) {
...
     function addFilter(d) {
-      filters.push(d);
-      notifyObservers();
-      d.setRefresh(refresh);
-      refresh();
+      var newItem = true;
+
+      filters.forEach(function (f) {
+        if (f.unique === d.unique) {
+          removeFilter(f);
```
Comment (genofire) [inline lib/datadistributor.js]: you remove the filter and disable add a new ...

File: lib/filters/genericnode.js (hunk 1 above)
Comment (genofire) [reply lib/filters/genericnode.js]: I am not sure, if a object equality behould be enought, i think not.

**Review by genofire — COMMENTED** [review]
If you would allow it, i will make a new commit with some tests/ideas?

File: lib/datadistributor.js (hunk 2 above)
Comment (genofire) [reply lib/datadistributor.js]: Ups sorry, you want toggle ...
:+1:

Maybe make a negativ filter?

File: lib/filters/genericnode.js (hunk 1 above)
Comment (genofire) [reply lib/filters/genericnode.js]: Like it is in `removeFilter`

**Comment by genofire** [comment]
Ups i am not really wake up.

File: lib/filters/genericnode.js (hunk 1 above)
Comment (xf-) [reply lib/filters/genericnode.js]: only setRefresh and render is available. I will add a function like getKey.

File: lib/datadistributor.js (hunk 2 above)
Comment (xf-) [reply lib/datadistributor.js]: i thought about that, but if you have 2 filters it would be possible to end up with no node = no filter left.

**Review by genofire — APPROVED** [review]

**Pull request merged**